### PR TITLE
Update agent images to debian8

### PR DESCRIPTION
**Pull request by fejta**
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/662)

### CODE REVIEW

**Comment by spxtr** [comment]
How do we test this?

File: jenkins/agent-ctl.sh [hunk 1]
```diff
@@ -52,15 +52,15 @@ usage() {
 set -o nounset
 set -o errexit
 
-DOCKER_VERSION='1.9.1-0~wheezy'
+DOCKER_VERSION='1.9.1-0~jessie'
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: would it be an unreasonable idea to update to 1.12 at the same time?

File: jenkins/agent-ctl.sh [hunk 2]
```diff
@@ -75,12 +75,21 @@ while true; do
...
+      IMAGE_FLAG='--image=jenkins-agent-20160613-1431'
+      PR=yes
+      ;;
     --pr)
       PR=yes
       shift
       ;;
     --base-image)
-      IMAGE_FLAG="--image=${BASE_IMAGE}"
+      IMAGE_FLAG="--image-family=${BASE_IMAGE}"
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: is the switch from `image` to `image-family` intentional?

I'm trying to understand this change wrt the documentation for this flag:
> --base-image: create instance with base image instead of family

File: jenkins/agent-ctl.sh [hunk 3]
```diff
@@ -75,12 +75,21 @@ while true; do
       FAKE=yes
       shift
       ;;
+    --previous)
```
Comment (ixdy) [inline jenkins/agent-ctl.sh]: need to document these in usage

Comment (fejta) [inline jenkins/agent-ctl.sh]: meh, I just want someplace to note the last version we were using. I think this is easier for someone reading the code to understand than `# previous --pr was jenkins-agent-20160613-2240` but would you prefer I do it that way?

File: jenkins/agent-ctl.sh (hunk 1 above)
Comment (fejta) [inline jenkins/agent-ctl.sh]: That is outside the scope of this PR.

Updating to debian8 should be a noop. Updating docker -- especially since we still use 1.9 in our [test images](https://github.com/kubernetes/test-infra/blob/master/jenkins/test-image/Dockerfile#L48) -- seems like a substantially more risky change.

This should happen once we have test images using different docker versions.

File: jenkins/agent-ctl.sh (hunk 2 above)
Comment (fejta) [inline jenkins/agent-ctl.sh]: Yes this is intentional. Updated the flag doc.

Now that image-families exist and are GA gcloud will drop support for `--image=debian-7-backports` (or `--image=debian-8`) being a special case alias for "list the current debian-7-backports and find the latest non-deprecated one."

base image in this case means the image we use to create the image (aka the `/debian-cloud/debian-8` image instead of the `jenkins-agent` image)

**Comment by fejta** [comment]
Joe I tested this by creating a couple PR agents with this image and ensuring that jobs pass. Then I updated all the pr agents to use this image.

I am waiting for 1.4 to ship before updating ci jenkins

File: jenkins/agent-ctl.sh (hunk 3 above)
Comment (spxtr) [inline jenkins/agent-ctl.sh]: I think it's fine as-is, but the "Usage" output should include what they mean.

**Comment by spxtr** [comment]
Looks good to me other than nit.

**Comment by ixdy** [comment]
@fejta can you address @spxtr's nit?

**Comment by fejta** [comment]
Yes I will get to it but I am buried in 1.5 planning and have had little coding time.

File: jenkins/agent-ctl.sh (hunk 3 above)
Comment (fejta) [inline jenkins/agent-ctl.sh]: Done
